src/common/charset/Charset.cpp

```diff
@@ -0,0 +1,196 @@
+ */
+
+#include "charset/Charset.h"
+
+namespace nebula {
+
+std::unordered_set<std::string> CharsetInfo::supportCharset = {"utf8"};
+
+std::unordered_set<std::string> CharsetInfo::supportCollation = {"utf8_bin"};
+
+std::unordered_map<std::string, CharsetInfo::CharsetToCollation> CharsetInfo::charsetToCollation = {
+    {"utf8", {"utf8", "utf8_bin", {"utf8_bin"}, "UTF-8 Unicode", 4}}
+};
+
+std::unordered_map<std::string, std::string> CharsetInfo::collateToLocale = {
+    {"utf8_bin", "en_US.UTF-8"}
+};
+
+Status CharsetInfo::isSupportCharset(const std::string& charsetName) {
+    for (auto& sc : supportCharset) {
+        if (!sc.compare(charsetName)) {
+            return Status::OK();
+        }
+    }
+    return Status::Error("Charset `%s' not support", charsetName.c_str());
+}
+
+
+Status CharsetInfo::isSupportCollate(const std::string& collateName) {
+    for (auto& sc : supportCollation) {
+        if (!sc.compare(collateName)) {
+            return Status::OK();
+        }
+    }
+    return Status::Error("Collation `%s' not support", collateName.c_str());
+}
+
+
+Status CharsetInfo::charsetAndCollateMatch(const std::string& charsetName,
+                                           const std::string& collateName) {
+    auto iter = charsetToCollation.find(charsetName);
+    if (iter != charsetToCollation.end()) {
+        for (auto& sc : iter->second.supportColl_) {
+            if (!sc.compare(collateName)) {
+                return Status::OK();
+            }
+        }
+    }
+    return  Status::Error("Charset `%s' and Collation `%s' not match",
+                          charsetName.c_str(), collateName.c_str());
+}
+
+
+StatusOr<std::string> CharsetInfo::getDefaultCollationbyCharset(const std::string& charsetName) {
+    auto iter = charsetToCollation.find(charsetName);
+    if (iter != charsetToCollation.end()) {
+        return iter->second.defaultColl_;
+    }
+    return Status::Error("Charset `%s' not support", charsetName.c_str());
+}
+
+
+StatusOr<std::string> CharsetInfo::getCharsetbyCollation(const std::string& collationName ) {
+    for (auto& cset : charsetToCollation) {
+        for (auto& coll : cset.second.supportColl_) {
+            if (!coll.compare(collationName)) {
+                return cset.first;
+            }
+        }
+    }
+    return Status::Error("Collation `%s' not support", collationName.c_str());
+}
+
+
+size_t CharsetInfo::getUtf8Charlength(const std::string& str) {
+    size_t length = 0;
+    for (size_t i = 0, len = 0; i < str.length(); i += len) {
+        unsigned char byte = str[i];
+        if (byte >= 0xFC) {
+            len = 6;
+        } else if (byte >= 0xF8) {
+            len = 5;
+        } else if (byte >= 0xF0) {
+            len = 4;
+        } else if (byte >= 0xE0) {
+            len = 3;
+        } else if (byte >= 0xC0) {
+            len = 2;
+        } else {
+            len = 1;
+        }
+        length++;
+    }
+    return length;
+}
```

Comment:
A question, this method same with MultiByteToWideChar?

Reply:
This function counts how many characters are in the string according to the UTF8 character encoding